File: Project2/Guarana.cpp
#include "Guarana.h"
#include "World.h"
#include <stdlib.h>

Guarana::Guarana(const Vector2& pos) : Plant(pos, 0, 15, 'g')
{
}

Guarana::~Guarana()
{
}

void Guarana::Hit(Organism* attacker)
{
	Plant::Hit(attacker);
	attacker->SetStrength(attacker->GetStrength() + 3);
	world->Logs() << GetName() << " made " << attacker->GetName() << " stronger [" << attacker->GetStrength() << "]\n";
}

Guarana* Guarana::Clone(const Vector2& pos) const
{
	return new Guarana(pos);
}

std::string Guarana::GetName() const
{
	return "Guarana";
}

File: Project2/WorldSimulator/Plants/Guarana.java
package WorldSimulator.Plants;

import WorldSimulator.Organism;
import WorldSimulator.Plant;
import WorldSimulator.Vector2;

import java.awt.*;

public class Guarana extends Plant
{
    public Guarana(Vector2 pos)
    {
        super(pos, 0, 15, 'g', new Color(18, 93, 231));
    }

    @Override
    protected void Hit(Organism attacker)
    {
        super.Hit(attacker);
        attacker.setStrength(attacker.getStrength() + 3);
        world.AddLog(this + " made " + attacker + " stronger [" + attacker.getStrength() + "]");
    }

    @Override
    public String toString()
    {
        return "Guarana";
    }
}
